mm2src/coins/tendermint/tendermint_tx_history_v2.rs

```diff
@@ -0,0 +1,819 @@
+                    };
+
+                    let fee_amount_with_denom = format!("{}{}", fee_data.amount, fee_data.denom);
+
+                    let transfer_details_list = get_transfer_details(tx.tx_result.events, fee_amount_with_denom);
+
+                    if transfer_details_list.is_empty() {
+                        log::debug!(
+                            "Could not find transfer details in events for tx '{}', skipping it",
+                            &tx_hash
+                        );
+                        continue;
+                    }
+
+                    let fee_details = try_or_continue!(
+                        get_fee_details(deserialized_tx.auth_info.fee, coin),
+                        "get_fee_details failed"
+                    );
+
+                    let mut fee_added = false;
+                    for (index, transfer_details) in transfer_details_list.iter().enumerate() {
+                        let mut internal_id_hash = index.to_le_bytes().to_vec();
+                        internal_id_hash.extend_from_slice(tx_hash.as_bytes());
+                        drop_mutability!(internal_id_hash);
+
+                        let internal_id = H256::from(internal_id_hash.as_slice()).reversed().to_vec().into();
+
+                        if let Ok(Some(_)) = storage
+                            .get_tx_from_history(&coin.history_wallet_id(), &internal_id)
+                            .await
+                        {
+                            log::debug!("Tx '{}' already exists in tx_history. Skipping it.", &tx_hash);
+                            continue;
+                        }
+
+                        let tx_sent_by_me = address.clone() == transfer_details.from;
+                        let is_platform_coin_tx = transfer_details.denom == coin.platform_denom();
+                        let is_self_tx = transfer_details.to == transfer_details.from;
+                        let mut tx_amounts = get_tx_amounts(transfer_details.amount, tx_sent_by_me, is_self_tx);
+
+                        if !fee_added
+                        // if tx is platform coin tx and sent by me
+                            && is_platform_coin_tx && tx_sent_by_me && !is_self_tx
+                        {
+                            tx_amounts.total += BigDecimal::from(fee_details.uamount);
+                            tx_amounts.spent_by_me += BigDecimal::from(fee_details.uamount);
+                            fee_added = true;
+                        }
+                        drop_mutability!(tx_amounts);
+
+                        let token_id: Option<BytesJson> = match !is_platform_coin_tx {
+                            true => {
+                                let denom_hash = sha256(transfer_details.denom.clone().as_bytes());
+                                Some(H256::from(denom_hash.as_slice()).to_vec().into())
+                            },
+                            false => None,
+                        };
```

Comment:
How about using `if else` statements? I find them more clear for matching bool values.
Else I think this
```rust
if is_platform_coin_tx {
  None
} else {
  let denom_hash = sha256(transfer_details.denom.clone().as_bytes());
  Some(H256::from(denom_hash.as_slice()).to_vec().into())
}
```
is clearer than
```rust
if !is_platform_coin_tx {
  let denom_hash = sha256(transfer_details.denom.clone().as_bytes());
  Some(H256::from(denom_hash.as_slice()).to_vec().into())
} else {
  None
}
```

Reply:
I guess this is pretty relative. Because `rustfmt` formats `if-else` statements with brackets in between them, match statement looks more readable to me. Also doing the action directly on the first condition makes more sense in my opinion(like in your first example, not the second one).

I can update this to anything if you want. I am okay with all of them.